### PR TITLE
[SPARK-41422][UI] Protobuf serializer for ExecutorSummaryWrapper

### PR DESCRIPTION
**Pull request by techaddict**
### What changes were proposed in this pull request?
Add Protobuf serializer for ExecutorSummaryWrapper

### Why are the changes needed?
Support fast and compact serialization/deserialization for ExecutorSummaryWrapper over RocksDB.

### Does this PR introduce any user-facing change?
No

### How was this patch tested?
New UT

### CODE REVIEW

**Review by LuciferYang — APPROVED** [review]
+1, LGTM

**Comment by techaddict** [comment]
@gengliangwang this is ready for review

File: core/src/main/scala/org/apache/spark/status/protobuf/ExecutorSummaryWrapperSerializer.scala [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+      totalShuffleRead = binary.getTotalShuffleRead,
+      totalShuffleWrite = binary.getTotalShuffleWrite,
+      isBlacklisted = binary.getIsBlacklisted,
+      maxMemory = binary.getMaxMemory,
+      addTime = new Date(binary.getAddTime),
+      removeTime = removeTime,
+      removeReason = removeReason,
+      executorLogs = binary.getExecutorLogsMap.asScala.toMap,
+      memoryMetrics = memoryMetrics,
+      // TODO: fix toInt
```
Comment (gengliangwang) [inline core/src/main/scala/org/apache/spark/status/protobuf/ExecutorSummaryWrapperSerializer.scala:138]: @techaddict There are some todos here. Will you fix them?

Comment (techaddict) [reply core/src/main/scala/org/apache/spark/status/protobuf/ExecutorSummaryWrapperSerializer.scala]: done, made stageIds int64

Comment (techaddict) [reply core/src/main/scala/org/apache/spark/status/protobuf/ExecutorSummaryWrapperSerializer.scala]: @gengliangwang  done, made stageIds int64

**Review by gengliangwang — APPROVED** [review]

**Comment by gengliangwang** [comment]
Merging to master

**Pull request closed without merging**